cloud/profitbricks/profitbricks.py

```diff
@@ -549,24 +549,50 @@ def startstop_machine(module, profitbricks, state):
 
         if wait_timeout <= time.time():
             # waiting took too long
-            module.fail_json(msg = "wait for virtual machine state timeout on %s" % time.asctime())
+            module.fail_json(msg="wait for virtual machine state timeout on %s" % time.asctime())
 
     return (changed)
 
+
+def _get_datacenter_id(datacenters, identity):
+    """
+    Fetch and return datacenter UUID by datacenter name if found.
+    """
+    for datacenter in datacenters['items']:
+        if identity in (datacenter['properties']['name'], datacenter['id']):
+            return datacenter['id']
+    return None
+
+
+def _get_server_id(servers, identity):
+    """
+    Fetch and return server UUID by server name if found.
+    """
+    for server in servers['items']:
+        if identity in (server['properties']['name'], server['id']):
+            return server['id']
+    return None
+
+
 def main():
     module = AnsibleModule(
         argument_spec=dict(
             datacenter=dict(),
             name=dict(),
             image=dict(),
-            cores=dict(default=2),
-            ram=dict(default=2048),
-            volume_size=dict(default=10),
+            cores=dict(type='int', default=2),
+            ram=dict(type='int', default=2048),
+            cpu_family=dict(choices=['AMD_OPTERON', 'INTEL_XEON'],
+                            default='AMD_OPTERON'),
+            volume_size=dict(type='int', default=10),
+            disk_type=dict(choices=['HDD', 'SSD'], default='HDD'),
+            image_password=dict(default=None),
+            ssh_keys=dict(type='list', default=[]),
             bus=dict(default='VIRTIO'),
```

Comment:
could you please also add a choice list for bus?
~~~python
bus=dict(default='VIRTIO', choices=['IDE', 'VIRTIO']),
~~~